amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/WorkerStateChangeListener.java

```diff
@@ -27,4 +27,7 @@
 	}
 
 	void onWorkerStateChange(WorkerState newState);
+
+	default void onInitializationFailure(Throwable e) {
```

Comment:
Can you rename the method to something like `allInitializationAttemptsFailed(Throwable t)`, indicating that the handler is invoked after all the initialization attempt have failed.